<h2>Ejercicio 2<h2>

<h3>Inciso c<h3>

In [14]:
import pandas as pd
from wget import download
from os import path, listdir

In [15]:
#descargamos si no existen
if not path.exists("netflix_titles.txt"):
    download("https://ignaciorlando.github.io/datasets/data-science/netflix_titles.txt")
else:
    print("No vamos a bajar el archivo porque ya existe")

No vamos a bajar el archivo porque ya existe
No vamos a bajar el archivo porque ya existe


In [16]:
#leemos e imprimos el archovo txt
with open("netflix_titles.txt", "r", encoding="utf-8") as a:
    archivoTXT = a.read()

print(archivoTXT)

Películas y Series de TV de Netflix

Acerca del Conjunto de Datos

Netflix es una fuerza líder en el ámbito de los medios y la transmisión de video. Con una asombrosa variedad de más de 8,000 películas y programas de televisión accesibles en su plataforma, a mediados de 2021, su cantidad de suscriptores globales superaba los 200 millones. Este conjunto de datos tabulado cataloga exhaustivamente todas las ofertas en Netflix, incluidos detalles vitales como elenco, directores, calificaciones, año de lanzamiento, duración y más.

Descripción General del Conjunto de Datos:

El conjunto de datos de Títulos de Netflix es una compilación completa de películas y programas de televisión disponibles en Netflix, que cubre diversos aspectos como el tipo de título, director, elenco, país de producción, año de lanzamiento, calificación, duración, géneros (listados) y una breve descripción. Este conjunto de datos es fundamental para analizar tendencias en el contenido de Netflix, comprender la popula

<p>Tenemos que el CSV tiene columnas 'nulas' o vacias separadas por columnas. Ademas, tenemos valores como 'elenco' que tiene los actores separados
por coma pero esta bien porque estan encerrado entre "".
Para solucionar lo de las columnas vacias, detectamos solo las columnas validas que no son vacias.<p>

In [17]:
#cargamos el archivo especificando encoding latin-1
df = pd.read_csv("netflix_titles.csv", encoding="latin-1")

#eliminamos las columnas 'Unnamed' vacias generadas por comas
columnas_validas = [col for col in df.columns if not col.startswith("Unnamed")]
df = df[columnas_validas]

#cambiamos a UTF-8
df.to_csv("netflix_titles_limpio.csv", index=False, encoding="utf-8")

print(df)

     show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   
8807   s8808  TV Show     Parasyte: The Grey     Yeon Sang-ho   
8808   s8809    Movie                 Serena     Susanne Bier   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa 

In [18]:
#cabecera
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8809 entries, 0 to 8808
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8809 non-null   str  
 1   type          8809 non-null   str  
 2   title         8809 non-null   str  
 3   director      6175 non-null   str  
 4   cast          7984 non-null   str  
 5   country       7978 non-null   str  
 6   date_added    8799 non-null   str  
 7   release_year  8809 non-null   int64
 8   rating        8805 non-null   str  
 9   duration      8806 non-null   str  
 10  listed_in     8809 non-null   str  
 11  description   8809 non-null   str  
dtypes: int64(1), str(11)
memory usage: 826.0 KB


In [19]:
#primeros 5 registros
df.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


In [21]:
df['release_year'].median()

np.float64(2017.0)

In [20]:
#resumen estadistico
df.describe(include='all')

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
count,8809,8809,8809,6175,7984,7978,8799,8809.000000,8805,8806,8809,8809
unique,8809,2,8806,4529,7694,748,1768,NaN,18,220,516,8777
top,s1,Movie,15-Aug,Rajiv Chilaka,David Attenborough,United States,"January 1, 2020",NaN,TV-MA,1 Season,"Dramas, International Movies","Paranormal activity at a lush, abandoned prope..."
freq,1,6132,2,19,19,2819,109,NaN,3208,1794,362,4
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2014.181292,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,8.818932,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1925.000000,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2013.000000,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2017.000000,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2019.000000,NaN,NaN,NaN,NaN


<p>Podemos ver que faltan practicamente la totalidad de las columnas en el analisis, sucediendo esto porque justamente son columnas con tipo 'string'. Es decir, el analisis estadistico que se realiza esta disponible solo sobre las variables cuantitativas.
<p>Asi, incluimos la totalidad de las columnas, incluidas las cualitativas, mediante 'include = all' donde para estas mismas se incluye<p>

- unique: Especifica la cantidad de valores unicos.
- top: Especifica el valor moda.
- freq: Indica la frecuencia de la moda.

<h3>Inciso d<h3>

<p>Analisis<p>

- show_id: Representa un ID unico para cada pelicula cargada - Cuantitativa discreta.

- type: Representa el tipo de pelicula asociada pudiendo ser 'Movie' o 'TV show' - Cualitativa nominal - El tipo mas mencionado es 'TV show'.
- title: Representa el titulo de la pelicula - Cualitativa nominal.
- director: Representa el director de la pelicula - Cualitativa nominal - El director mas mencionado es 'Rajiv Chilaka'.
- cast: Representa el elenco de la pelicula - Cualitativa nominal - El actor mas mencionado es 'David Attenborough'.
- country: Representa el pais origen de la pelicula - Cualitativa nominal - El pais mas mencionado es 'United States'.
- date_added: Representa la fecha en que fue anadido los datos - Cualitativa ordinal - Se agruparon mas datos el 'January 1, 2020'.
- release_year: Representa el anio de lanzamiento de la pelicula - Cuantitativa discreta - La mediana es 2017.
- rating: Representa la clasificacion por edad y publico sugerido - Cualitativa ordinal - El rating mas mencionado es 'TV-MA'.
- duration: Representa la duracion del contenido - Cualitativo ordinal - La duracion mas frecuente es de '1 temporada'.
- listed_in: Representa las clasificaciones en cuanto a contenido - Cualitativo nominal - La clasificacion mas frecuente es 'Dramas, International Movies'.
- description: Representa la clasificacion del contenido - Cualitativo nominal.

<h3>Inciso e<h3>

In [ ]:
#iv
dfOld = df[df['release_year']==1925]
#o
top_5 = df.sort_values(by="release_year", ascending=True).head(1)
print(top_5)

In [30]:
#v
dfCountry = df[df['country']=='United States'].shape[0] #(0: filas, 1: columnas)
print(dfCountry)

12


i. 4529.

ii. Rajiv Chilaka.

iii. Tenemos 6132 programas de TV y entonces tenemos 2677 series.

iv. La pelicula mas vieja del catalogo es 'Pioneers: First Wowen Filmmakers' del anio 1925.

v. En bloque de arriba.